server/viame_tasks/tasks.py

```diff
@@ -150,3 +167,50 @@ def convert_video(self, path, folderId, token, auxiliaryFolderId):
         _file["itemId"], {"folderId": folderId, "codec": "h264"}
     )
     os.remove(output_path)
+
+
+@app.task(bind=True)
+def convert_images(self, folderId, token):
+    """
+    Ensures that all images in a folder are in a web friendly format (png or jpeg).
+
+    If conversions succeeds for an image, it will replace the image with an image
+    of the same name, but in a web friendly extension.
+
+    Returns the number of images successfully converted.
+    """
+    gc = self.girder_client
+    gc.token = token
+
+    items = gc.listItem(folderId)
+    skip_item = (
+        lambda item: item["name"].endswith(".png")
+        or item["name"].endswith(".jpeg")
+        or item["name"].endswith(".jpg")
+    )
+    valid_items = [item for item in items if not skip_item(item)]
+
+    count = 0
+    with tempfile.TemporaryDirectory() as temp:
+        dest_dir = Path(temp)
+
+        for item in valid_items:
+            # Assumes 1 file per item
+            gc.downloadItem(item["_id"], dest_dir, item["name"])
+
+            item_path = dest_dir / item["name"]
+            new_item_path = dest_dir / ".".join([*item["name"].split(".")[:-1], "png"])
+
+            process = Popen(
+                ["ffmpeg", "-i", item_path, new_item_path],
+                stderr=DEVNULL,
+                stdout=DEVNULL,
+            )
+
+            returncode = process.wait()
```

Comment:
did you have a reason for choosing `wait` over `communicate` and sending the logs to girder?  

Again, should probably do the same thing everywhere for consistency.  

I think it would actually be better to just let the logs flow to STDOUT for all the conversion tasks, so I'd opt for `wait`  everywher ffmpeg is used.,

Reply:
The reason I used `wait` is because we don't subsequently use the output in any way, but I wasn't aware of the fact that you can send logs to jobs. It seems that if we want to send these logs to the job (which I think we should), we'll have to use `process.communicate`, as simply letting them flow to stdout doesn't actually capture them (I suspect because they're run in their own process).